reference/best_practices/reusable_bundles.rst

```diff
@@ -1,4 +1,173 @@
+""""""""""""""""""""""""""""""""""""""""""""""""
+It is the best extensibility point you have.
+Unfortunately, it is not always possible as Akeneo PIM can't know where you want to plug your code.
+If you can not, it does not matter, you can decorate or extend your class and add the extension point you need.
+By the way, feel free to contribute opening an issue to discuss with our team on our `github repository`_
+or let us know your need on our `forum`_. It is important for us to improve ourselves and our application.
+
+.. _github repository: https://github.com/akeneo/pim-community-dev
+.. _forum: https://www.akeneo.com/fr/forums/
+
+
+Avoid to override Akeneo PIM classes/services
+"""""""""""""""""""""""""""""""""""""""""""""
+The easiest way to add a new behavior on your services is to override it.. But wait! That's not a good thing!
+First, each override leads to potential compatibility problems.
+Let's imagine that you override the `pim_catalog.saver.product` because you want to add a specific action when you save a product.
+Now, a final customer wants to use your bundle and another one which override the same service (Tough luck!).
+The last bundle defined in the `app/AppKernel.php` will defined the service used..
+And in both cases, one of the actions will not be applied.
+
+The second effect of overriding a parameter or a service is that it will have a side effect on the whole application.
+If you need to impact transversely Akeneo PIM and you can't plug on events, that's a way to do but keep in mind
+you can use create your own service with your own class.
+
+
+Models and Repositories
+"""""""""""""""""""""""
+You should avoid to extends the models (entities and documents) and repositories.
+The problem is the same than when you override classes/services.
+
+There is unfortunately no perfect solution for these cases.
+
+Don't extend an entity to add it properties.
+
+Two possibilities here:
+
+- There is not a lot of changes on the entity you overrode. Document what is missing and let's the integrators do it.
+- Do a `oneToOne unidirectional association`_. Here is an example with the `Category` entity. I want to add the description on it.
+  A way to do that avoiding to extend is to create a new entity named `MyExtendedCategory` with an id, a description and a relation to the `Category` entity.
+  The bad side of this solution is you won't have access to the description from the Category object (the reverse is possible).
+  It's sometimes frustrating to access it by a non-natural way.
+
+
+For your own model classes, create your class and its interface.
+Then you can rely on your interface and use the `Akeneo target resolver`_ which is based on the `Doctrine target entity resolver`_.
+
+
+About repositories, you can create and inject your own service. Doctrine does not allow to have many repositories.
+Don't tint it as a doctrine one, just inject the `ObjectManager` (Entity or Document) and the class you want to rely on.
```

Comment:
"To tint" is color related in English. You should say "For this reason (we cannot have multiple repositories per entity, ndlr), you can't declare them as Doctrine repository. yet you can create a service, inject ObjectManager and the class you want to rely on. Do you have any repository example where you do this ?